src/index.js

```diff
@@ -0,0 +1,358 @@
+  return {
+    public: buf2hex(publicKey),
+    private: buf2hex(privateKey),
+    chain: buf2hex(chainCode),
+  };
+};
+
+
+
+/**
+ * Wraps nacl.lowlvel.crypto_hash
+ * @param  {Uint8Array} seed
+ * @return {array}
+ */
+const naclHash = seed => {
+  let newHash = []
+  nacl.lowlevel.crypto_hash(newHash, seed.reverse(), seed.length)
+  return newHash
+}
+
+
+
+/**
+ * Derive Urbit network keypairs from a seed. Matches ++pit:nu:crub:crypto
+ * @param  {buffer} seed     seed to derive from
+ * @param  {string} password optional password to salt the seed before deriving
+ * @return {object} urbitKeys, derived according to ++pit:nu:crub:crypto.
+ */
+const urbitKeysFromSeed = (seed, password) => {
+  const h = naclHash(bufferConcat([seed, password]));
+
+  const c = h.slice(32);
+  const a = h.slice(0, 32);
+
+  const crypt = nacl.sign.keyPair.fromSeed(bufferFrom(c));
+  const auth = nacl.sign.keyPair.fromSeed(bufferFrom(a));
+
+  return {
+    crypt: {
+      private: buf2hex(reverse(c)),
+      public: buf2hex(reverse(crypt.publicKey)),
+    },
+    auth: {
+      private: buf2hex(reverse(a)),
+      public: buf2hex(reverse(auth.publicKey)),
+    }
+  };
+};
+
+
+
+/**
+ * Derive all keys from the ticket.
+ * @param  {string, Uint8Array, buffer}  ticket ticket, at least 16 bytes.
+ * @param  {integer}  seedSize desired size of the generated seeds in bytes.
+ * @param  {array of integers}  ships array of ship-numbers to generate keys for.
+ * @param  {string}  password optional password to use during derivation.
+ * @param  {object}  revisions optional revision per key purpose:
+ * (transfer, spawn, delegate, manage, network), defaults to all-zero
+ * @return {Promise => object} an object representing a full HD wallet.
+ */
+const fullWalletFromTicket = async config => {
+  const { ticket, seedSize, ships, password, revisions } = config;
+  const seed = await argon2u(ticket, seedSize).hash;
+  return fullWalletFromSeed(bufferFrom(seed), ships, password, revisions);
+}
+
+
+
+/**
+ * Derive all keys from a seed.
+ * @param  {string, Uint8Array, buffer}  ownerSeed ticket, at least 16 bytes.
+ * @param  {array of integers}  ships array of ship-numbers to generate keys for.
+ * @param  {string}  password optional password to use during derivation.
+ * @param  {object}  revisions optional revision per key purpose:
+ * (transfer, spawn, delegate, manage, network), defaults to all-zero
+ * @return {Promise => object} an object representing a full HD wallet.
+ */
+const fullWalletFromSeed = async config => {
+  const { ownerSeed, ships, password, revisions } = config;
+
+  // Normalize revisions object
+  const _revisions = {
+    transfer: get(revisions, 'transfer', 0),
+    spawn: get(revisions, 'spawn', 0),
+    delegate: get(revisions, 'delegate', 0),
+    manage: get(revisions, 'manage', 0),
+    network: get(revisions, 'network', 0),
+  };
+
+  const ownershipNode = await childNodeFromSeed({
```

Comment:
@gavinpatkinson Actually, I realize now that this isn't correct, and doesn't match the original implementation. What you're doing here is deriving a seed from the `ownerSeed`, and then getting a wallet from that. What you really want to be doing (according to previous implementation and design) is to derive a wallet from the `ownerSeed` directly.